moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/trajectory_monitor.h

```diff
@@ -96,6 +115,7 @@ class TrajectoryMonitor
   void recordStates();
 
   CurrentStateMonitorConstPtr current_state_monitor_;
+  std::unique_ptr<MiddlewareHandle> middleware_handle_;
```

Comment:
@tylerjw where is moveit style on class member documentation.
https://moveit.ros.org/documentation/contributing/code/ implies http://wiki.ros.org/CppStyleGuide#Documentation to me
> All functions, methods, classes, class variables, enumerations, and constants should be documented.

Reply:
I would specifically disagree with the idea of documenting class variables.  I would prefer we focus on documenting the public interface.  I'm not sure what a useful comment on this line would be?

Reply:
Consider
```
CurrentStateMonitorConstPtr current_state_monitor_;  //!< Samples joint states.
std::unique_ptr<MiddlewareHandle> middleware_handle_;  //!< Interface for communicating with ROS.
```
It may not be part of the public interface, but it is a public facing member.

Reply:
Also, what do you mean by "focus on documenting the public interface"? Everything public facing should be documented imho

Reply:
You are correct that these are not in an impl class.  I would still however focus on documenting the parts of a class that the user can access without inheriting or making it a friend class first (that is what I meant by public interface).

Reply:
MoveIt is not properly pimpl'ed (even in the few places where it is it's somewhat smelly), otherwise this would be a different matter. As MoveIt is generally extendable/hackable on any level, I'm no big fan of having pimpl/private declarations in the framework for most members anyway, but that's an independent discussion.

Either way, adding brief inline comments makes sense when they add additional information and adding them takes less resources than discussing in long threads whether focusing on other things makes more sense :smirk: 
I don't feel the proposed comments add relevant information, but then that's usually just maintainer bias.
 `[csm] //!< Samples robot states.` is somewhat more accurate, because CSM also updates from TF where required.

Reply:
> even in the few places where it is it's somewhat smelly
:smirk: 

I probably wouldn't advocate for arbitrary use of pimpl (Especially since I generally advocate against pointers where dynamic memory isn't required. Additionally I think ROS developers over use pointers because that's what they were taught).
I know it has a purpose in maintaining a stable ABI.
I am a big fan of composition, though not simply using pimpl without the "p".

> Either way, adding brief inline comments makes sense when they add additional information

Mainly I'm just I'm just trying to clarify a point of policy/bike shedding, since the moveit style guide said/implied that it follows the ROS style guide. At previous companies, we documented every member and interface with no ill effects. One might even argue that if you can't come up with a logical and simple comment for the member, perhaps it shouldn't be there.
Regardless it seems that the style that member comments are optional. Perhaps I should update the style webpage @tylerjw? I wonder what else deviates.

> [csm] //!< Samples robot states. is somewhat more accurate, because CSM also updates from TF where required.

This was mainly a dig to @tylerjw from last week at our hackathon where we started this since every one was making fun of the name and didnt' **really** know what the class did until we read the entire implementation.

Reply:
> Additionally I think ROS developers over use pointers because that's what they were taught

Not just taught, but also forced by existing shared_ptr APIs. Fully agreed. :worried: 

> One might even argue that if you can't come up with a logical and simple comment for the member, perhaps it shouldn't be there.

One might just as well argue that if you can't come up with a comment that's more descriptive than the variable/class name, it might just be that the author did a good job naming them :cat2: 

> we documented every member and interface with no ill effects

Adding explanations always makes sense. Period. Adding comments for syntactic reasons, on the other hand, if they don't do more than repeat the identifier, amounts to a `NOLINT` flag for the line, so reviewers don't complain about missing comments. :)
It's hard to find the line between `if (exists) { // if it exists` and `CurrentStateMonitor csm; //!< Samples states`, especially when you're working with the code for years.